piff/gp_interp.py

```diff
@@ -0,0 +1,324 @@
+
+        :param star:        A Star instance to which one wants to interpolate
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        :returns: a new Star instance with its StarFit member holding the interpolated parameters
+        """
+        # because of sklearn formatting, call interpolateList and take 0th entry
+        return self.interpolateList([star], logger=logger)[0]
+
+    def interpolateList(self, stars, logger=None):
+        """Perform the interpolation for a list of stars.
+
+        :param star_list:   A list of Star instances to which to interpolate.
+        :param logger:      A logger object for logging debug info. [default: None]
+
+        :returns: a list of new Star instances with interpolated parameters
+        """
+        Xstar = np.array([self.getProperties(star) for star in stars])
+        y = self._predict(Xstar)
+        fitted_stars = []
+        for y0, star in zip(y, stars):
+            if star.fit is None:
+                fit = StarFit(y)
+            else:
+                fit = star.fit.newParams(y0)
+            fitted_stars.append(Star(star.data, fit))
+        return fitted_stars
+
+    def _finish_write(self, fits, extname):
+        # Note, we're only storing the training data and hyperparameters here, which means the
+        # Cholesky decomposition will have to be re-computed when this object is read back from
+        # disk.
+        init_theta = self.kernel.theta
+        fit_theta = self.gp.kernel_.theta
+        dtypes = [('INIT_THETA', init_theta.dtype, init_theta.shape),
+                  ('FIT_THETA', fit_theta.dtype, fit_theta.shape),
+                  ('X', self._X.dtype, self._X.shape),
+                  ('Y', self._y.dtype, self._y.shape)]
+
+        data = np.empty(1, dtype=dtypes)
+        data['INIT_THETA'] = init_theta
+        data['FIT_THETA'] = fit_theta
+        data['X'] = self._X
+        data['Y'] = self._y
+
+        fits.write_table(data, extname=extname+'_kernel')
+
+    def _finish_read(self, fits, extname):
+        data = fits[extname+'_kernel'].read()
+        optimizer = self.gp.optimizer
+
+        # Run fit to set up GP, but don't actually do any hyperparameter optimization.  Just
+        # set the GP up using the current hyperparameters.
+        self.gp.kernel.theta = data['FIT_THETA'][0]
+        optimizer = self.gp.optimizer
+        self.gp.optimizer = None
+        self._fit(data['X'][0], data['Y'][0])
+        self.gp.optimizer = optimizer
+        # Now that gp is setup, we can restore it's initial kernel.
+        self.gp.kernel.theta = self.kernel.theta = data['INIT_THETA'][0]
+
+
+
+from sklearn.gaussian_process.kernels import StationaryKernelMixin, NormalizedKernelMixin, Kernel
+from sklearn.gaussian_process.kernels import Hyperparameter
+
+class EmpiricalKernel(StationaryKernelMixin, NormalizedKernelMixin, Kernel):
```

Comment:
Why is this called `Empirical`?  That implies to me something based on the data.  Not a kernel with a given explicit function.  It seems like `FunctionKernel` or `ExplicitKernel` would be more apt.